src/site/shared/containers/HistoryBox/index.tsx

```diff
@@ -15,49 +15,80 @@ import "./index.scss";
     g: GroupAction;
+    isRedo: boolean;
 }
-const GroupActionEntry = ({ g }: GroupActionEntryProps) => {
+const GroupActionEntry = ({ g, isRedo }: GroupActionEntryProps) => {
     const [isCollapsed, setIsCollapsed] = useState(true);
 
     if (g.isEmpty())
         return null;
     if (g.getActions().length === 1)
-        return (<HistoryEntry a={g.getActions()[0]}></HistoryEntry>);
-
-    return (
-        <div className="historybox__groupentry"
-             onClick={(e) => {
-                 // Necessary to stop child entries from collapsing the parent history entry
-                 e.stopPropagation();
-                 setIsCollapsed(!isCollapsed);
-             }}>
-            <span>{g.getName()}</span>
-            <span
-                className={`historybox__groupentry__collapse_btn \
-                            ${isCollapsed ? "historybox__groupentry__collapse_btn-collapsed" : "" }`}>
-                &rsaquo;
-            </span>
-            {!isCollapsed && g.getActions().map((a, i) => {
-                return(<HistoryEntry key={`group-action-entry-${i}`} a={a}></HistoryEntry>);
-            })}
-        </div>
-    );
+        return (<HistoryEntry a={g.getActions()[0]} isRedo={isRedo}></HistoryEntry>);
+    if(!isRedo)
+        return (
+            <div className="historybox__groupentry"
+                onClick={(e) => {
+                    // Necessary to stop child entries from collapsing the parent history entry
+                    e.stopPropagation();
+                    setIsCollapsed(!isCollapsed);
+                }}>
+                <span>{g.getName()}</span>
+                <span
+                    className={`historybox__groupentry__collapse_btn \
+                                ${isCollapsed ? "historybox__groupentry__collapse_btn-collapsed" : "" }`}>
+                    &rsaquo;
+                </span>
+                {!isCollapsed && g.getActions().map((a, i) => {
+                    return(<HistoryEntry key={`group-action-entry-${i}`} a={a} isRedo={false} ></HistoryEntry>);
+                })}
+            </div>
+        );
+    else
+        return (
+            <div className="historybox__dashedgroupentry"
+                onClick={(e) => {
+                    // Necessary to stop child entries from collapsing the parent history entry
+                    e.stopPropagation();
+                    setIsCollapsed(!isCollapsed);
+                }}>
+                <span>{g.getName()}</span>
+                <span
+                    className={`historybox__groupentry__collapse_btn \
+                                ${isCollapsed ? "historybox__groupentry__collapse_btn-collapsed" : "" }`}>
+                    &rsaquo;
+                </span>
+                {!isCollapsed && g.getActions().map((a, i) => {
+                    return(<HistoryEntry key={`group-action-dashedentry-${i}`} a={a} isRedo={true} ></HistoryEntry>);
+                })}
+            </div>
+        );
```

Comment:
This can probably be condensed and just use a ternary operator for the first className and the key, since it seems like those are the only differences between the if and the else